File: js/programmers/level1/폰켓몬.js
function solution(nums) {
  const canTakeCnt = nums.length / 2;
  const set = new Set(nums);
  return set.size > canTakeCnt ? canTakeCnt : set.size;
}

console.log(solution([3, 1, 2, 3]));
console.log(solution([3, 3, 3, 2, 2, 4]));
console.log(solution([3, 3, 3, 2, 2, 2]));

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        Set<Integer> s = new HashSet();
        for (int num : nums) {
            s.add(num);
        }
        
        return s.size() > nums.length / 2 ? nums.length / 2 : s.size();
    }
}